lib/validate.js

```diff
@@ -9,19 +9,22 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-/** @private */
-function validateName (input) {
-  const valid = /^[a-zA-Z0-9]{1,20}$/
-  return valid.test(input) || `'${input}' is not a valid name for creation, the name must match /^[a-zA-Z0-9]{1,20}$/`
-}
-
-/** @private */
-function validateTitle (input) {
-  const valid = /^[\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{1,45}$/
-  return valid.test(input) || `'${input}' is not a valid title for creation, the name must match /^[\\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{1,45}$/`
+/**
+ * @private
+ */
+function createValidator (name, regex) {
+  return function (input) {
+    return regex.test(input) || `'${input}' is not a valid ${name}, it must match ${regex.toString()}`
+  }
 }
 
 module.exports = {
-  validateName,
-  validateTitle
+  // project input validators
+  validateProjectName: createValidator('Project name', /^[a-zA-Z0-9]{1,20}$/),
+  validateProjectTitle: createValidator('Project title', /^[\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{1,45}$/),
+  validateProjectDescription: createValidator('Project description', /^.{0,1000}$/), // can be empty
+  // workspace input validators
+  validateWorkspaceName: createValidator('Workspace name', /^[a-zA-Z0-9]{1,20}$/),
+  validateWorkspaceTitle: createValidator('Workspace title', /^[\sA-Za-z0-9\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff]{0,45}$/) // can be empty
+  // validateWorkspaceDescription: createValidator('Workspace description', /^(?=\S).{0,500}(?<=\S)$/)
```

Comment:
i assume there is no restrictions for description?

Reply:
for projects is 1000 chars max, and 500 for workspaces however we don't asked for a description in workspaces, should we ?

Reply:
true, so no need for validation there